cmake/FindWin32SslRuntime.cmake

```diff
@@ -5,22 +5,37 @@ if (WIN32)
   # Get standard installation paths for OpenSSL under Windows 
 
   # http://www.slproweb.com/products/Win32OpenSSL.html
-  set(_OPENSSL_ROOT_HINTS
-    ${OPENSSL_ROOT_DIR}
-    "[HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\OpenSSL (32-bit)_is1;Inno Setup: App Path]"
-    "[HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\OpenSSL (64-bit)_is1;Inno Setup: App Path]"
-    ENV OPENSSL_ROOT_DIR
-    )
-  file(TO_CMAKE_PATH "$ENV{PROGRAMFILES}" _programfiles)
-  set(_OPENSSL_ROOT_PATHS
-    "${_programfiles}/OpenSSL"
-    "${_programfiles}/OpenSSL-Win32"
-    "${_programfiles}/OpenSSL-Win64"
-    "C:/OpenSSL/"
-    "C:/OpenSSL-Win32/"
-    "C:/OpenSSL-Win64/"
-    )
-  unset(_programfiles)
+
+  if( CMAKE_SIZEOF_VOID_P EQUAL 8 )
+    # target win64
+    set(_OPENSSL_ROOT_HINTS
+      ${OPENSSL_ROOT_DIR}
+      "[HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\OpenSSL (64-bit)_is1;Inno Setup: App Path]"
+      ENV OPENSSL_ROOT_DIR
+      )
+    file(TO_CMAKE_PATH "$ENV{PROGRAMFILES}" _programfiles)
+    set(_OPENSSL_ROOT_PATHS
+      "${_programfiles}/OpenSSL-Win64"
+      "C:/OpenSSL-Win64/"
+      )
+    unset(_programfiles)
+  else( CMAKE_SIZEOF_VOID_P EQUAL 8 )

```

Comment:
They guys developing cmake originally made some weird syntax choices, like this one. We already define `CMAKE_ALLOW_LOOSE_LOOP_CONSTRUCTS`, so we could remove the condition from else() and endif()